tools/azure-sdk-tools/devtools_testutils/proxy_testcase.py

```diff
@@ -0,0 +1,147 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+import os
+import requests
+
+try:
+    # py3
+    import urllib.parse as url_parse
+except:
+    # py2
+    import urlparse as url_parse
+
+import subprocess
+
+# the functions we patch
+from azure.core.pipeline.transport import RequestsTransport
+
+# the trimming function to clean up incoming arguments to the test function we are wrapping
+from azure_devtools.scenario_tests.utilities import trim_kwargs_from_test_function
+
+# defaults
+PROXY_URL = "http://localhost:5000"
```

Comment:
If we're ok with manual start of the proxy, we should probably assume SSL. I'll see if I can figure that out today. If I can, let's get those changes added to this PR and swap this to https.